sdk/servicebus/microsoft-azure-servicebus/src/main/java/com/microsoft/azure/servicebus/MessageConverter.java

```diff
@@ -131,14 +131,20 @@ public static Message convertAmqpMessageToBrokeredMessage(org.apache.qpid.proton
         // Header
         // Delivery count for service bus starts from 1, for AMQP it starts from 0.
         brokeredMessage.setDeliveryCount(amqpMessage.getDeliveryCount() + 1);
-        brokeredMessage.setTimeToLive(Duration.ofMillis(amqpMessage.getTtl()));
-        
+        long ttlMillis = amqpMessage.getTtl();
+        if (ttlMillis > 0l) {
+        	brokeredMessage.setTimeToLive(Duration.ofMillis(ttlMillis));
+        }
 
         // Properties
         // Override TimeToLive from CrationTime and ExpiryTime, as they support duration of any length, which ttl doesn't
         if (amqpMessage.getCreationTime() != 0l && amqpMessage.getExpiryTime() != 0l) {
-        	brokeredMessage.setTimeToLive(Duration.ofMillis(amqpMessage.getExpiryTime() - amqpMessage.getCreationTime()));
+        	ttlMillis = amqpMessage.getExpiryTime() - amqpMessage.getCreationTime();
```

Comment:
I understand that this override is done for a long time ,  Why is it done in first place ? If  `amqpMessage.getTtl()`    is positive , this override is needed ?

Reply:
Needed because ttl is milliseconds as integer, so it can't support time to live of more than 50 days. That's why we had to change it.